pennylane/io.py

```diff
@@ -410,113 +407,140 @@ def from_qiskit_op(qiskit_op, params=None, wires=None):
         measurements (None | MeasurementProcess | list[MeasurementProcess]): an optional PennyLane
-            measurement or list of PennyLane measurements that overrides any terminal measurements
-            that may be present in the input circuit. If set to ``None``, existing measurements
-            in the input circuit will be used.
+            measurement or list of PennyLane measurements that overrides the terminal measurements
+            that may be present in the input circuit. Defaults to ``None``, such that all existing measurements
+            in the input circuit are returned. See **Removing terminal measurements** for details.
 
     Returns:
-        function: the PennyLane template created based on the QASM string
+        function: the PennyLane quantum function created based on the QASM string. This function itself returns the mid-circuit measurements plus the terminal measurements by default (``measurements=None``), and returns **only** the measurements from the ``measurements`` argument otherwise.
 
-    By default, ``from_qasm`` will remove any measurements that are present in the QASM code.
-    If the QASM code contains measurements, set ``measurements=None`` to keep them in the
-    output of ``from_qasm``.
+    **Example:**
 
-    .. warning::
+    .. code-block:: python
 
-        The current default behaviour of removing measurements in the QASM code is deprecated
-        and will be changed in a future release. Starting in version ``0.38``, ``from_qasm``
-        will keep the measurements from the QASM code by default. To remove all measurements,
-        set ``measurements=[]`` which overrides the existing measurements with an empty list.
+        qasm_code = 'OPENQASM 2.0;'\\
+                    'include "qelib1.inc";'\\
+                    'qreg q[2];'\\
+                    'creg c[2];'\\
+                    'h q[0];'\\
+                    'measure q[0] -> c[0];'\\
+                    'rz(0.24) q[0];'\\
+                    'cx q[0], q[1];'\\
+                    'measure q -> c;'
 
-    **Example:**
+        loaded_circuit = qml.from_qasm(qasm_code)
 
-    .. code-block:: python
+    >>> print(qml.draw(loaded_circuit)())
+    0: ──H──┤↗├──RZ(0.24)─╭●──┤↗├─┤
+    1: ───────────────────╰X──┤↗├─┤
+
+    Calling the quantum function returns a tuple containing the mid-circuit measurements and the terminal measurements.
+
+    >>> loaded_circuit()
+    (MeasurementValue(wires=[0]),
+    MeasurementValue(wires=[0]),
+    MeasurementValue(wires=[1]))
+
+    A list of measurements can also be passed directly to ``from_qasm`` using the ``measurements`` argument, making it possible to create a PennyLane circuit with :class:`qml.QNode <pennylane.QNode>`.
+
+        .. code-block:: python
+
+            dev = qml.device("default.qubit")
+            measurements = [qml.var(qml.Y(0))]
+            circuit = qml.QNode(qml.from_qasm(qasm_code, measurements = measurements), dev)
```

Comment:
The indentation is off here and its making these two code blocks (the one above the comment and the one below it) render in side a blue box

Reply:
I was literally wondering last friday if my eyes were just tired and that's why i saw the blue box. Thank you!